fabric-particles-v1/src/main/java/net/fabricmc/fabric/api/client/particles/FabricSpriteProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.particles;
```

Comment:
The package should use singular "particle" instead of "particles" and carry the module api version (v1) after the module name. In this file that'd be net.fabricmc.fabric.api.client.particle.v1

Reply:
> carry the module api version (v1) after the module name. 

Since when has this being a requirement? Including versions in the package automatically guarantees no backwards compatibility.

Reply:
Typically v1 and v2 would coexist for some period of time with v1 in a deprecated state. 

Changes that can be backwards compatible ( generally additive changes) would not prompt a major version change.

Reply:
Typically v1 and v2 would coexist for some period of time with v1 in a deprecated state. 

Changes that can be backwards compatible ( generally additive changes) would not prompt a major version change.

Reply:
Y'all already have such a long-ass package name, though. Sure, I'll change them. Just have to start up Eclipse.